NoMongo: Restore the GraphQL Introspection GitHub Action Test

Priyanshuthapliyal2005 opened the pull request:
<!--
This section can be deleted after reading.

We employ the following branching strategy to simplify the development process and to ensure that only stable code is pushed to the `master` branch:

- `develop`: For unstable code: New features and bug fixes.
- `master`: Where the stable production ready code lies. Only security related bugs.

NOTE!!!

ONLY SUBMIT PRS AGAINST OUR `DEVELOP` BRANCH. THE DEFAULT IS `MAIN`, SO YOU WILL HAVE TO MODIFY THIS BEFORE SUBMITTING YOUR PR FOR REVIEW. PRS MADE AGAINST `MAIN` WILL BE CLOSED.

-->

<!--
Thanks for submitting a pull request! Please provide enough information so that others can review your pull request.
-->

**What kind of change does this PR introduce?**
NoMongo: Restore the GraphQL Introspection GitHub Action Test
<!-- E.g. a bugfix, feature, refactoring, etc… -->

**Issue Number:**

Fixes #3235  <!--Add related issue number here.-->

**Snapshots/Videos:**

<!--Add snapshots or videos wherever possible.-->

**If relevant, did you update the documentation?**

<!--Add link to Talawa-Docs.-->

**Summary**

<!-- Explain the **motivation** for making this change. What existing problem does the pull request solve? -->
<!-- Try to link to an open issue for more information. -->

**Does this PR introduce a breaking change?**

<!-- If this PR introduces a breaking change, please describe the impact and a migration path for existing applications. -->

## Checklist

### CodeRabbit AI Review
- [x] I have reviewed and addressed all critical issues flagged by CodeRabbit AI
- [x] I have implemented or provided justification for each non-critical suggestion
- [x] I have documented my reasoning in the PR comments where CodeRabbit AI suggestions were not implemented

### Test Coverage
- [ ] I have written tests for all new changes/features
- [ ] I have verified that test coverage meets or exceeds 95%
- [ ] I have run the test suite locally and all tests pass


**Other information**

<!--Add extra information about this PR here-->

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**

<!--Yes or No-->

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **Workflow Updates**
  - Added a new job, `Graphql-Inspector`, to validate the GraphQL schema in the pull request workflow.
  - Introduced validation step for checking the existence of TypeScript files in the `src/GraphQl` directory.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

palisadoes reviewed (changes requested):
1. Reset your edit to the countline.py script. It is not in scope of the PR. 
2. Please don't edit code that is not relevant to the issue

Priyanshuthapliyal2005 commented:
> 1. Reset your edit to the countline.py script. It is not in scope of the PR.
> 2. Please don't edit code that is not relevant to the issue

Sorry for that change. Yesterday, I encountered a check failure because some files exceeded the 600-line limit.

Priyanshuthapliyal2005 commented on .github/workflows/pull-request.yml:
```diff
@@ -272,35 +272,71 @@
+
+      - name: resolve dependency
+        run: npm install -g @graphql-inspector/cli@latest
       
```
![image](https://github.com/user-attachments/assets/843e6af7-0440-46db-aa53-e211c14e97ad)
@coderabbitai The entire repo is using Node 22.x, as you can verify in the documentation and this screenshot. So, this suggestion is not valid.

Priyanshuthapliyal2005 commented:
@coderabbitai full review

palisadoes commented on .github/workflows/pull-request.yml:
```diff
@@ -272,35 +272,78 @@ jobs:
+          npm install -g @graphql-inspector/cli@latest
+
+          # Check if any .graphql files exist
+          if ! find src -name "*.graphql" -type f | grep -q .; then
```
This isn't required. We are only interested in schema.graphql

palisadoes commented on .github/workflows/pull-request.yml:
```diff
@@ -272,35 +272,78 @@ jobs:
+            echo "Warning: No .graphql files found in src directory"
+            exit 0
+          fi
+          # Check if schema.graphql exists
```
Why is this code in the testing application section and not in the GraphQL inspector section?

palisadoes commented on .github/workflows/pull-request.yml:
```diff
@@ -272,35 +272,78 @@ jobs:
+          fi
+          npx graphql-inspector validate schema.graphql "src/**/*.graphql"
+  Graphql-Inspector:
+    if: ${{ github.actor != 'dependabot[bot]' }}
```
This should depend on the tests being successful and not run independently. Use a `needs` statement

palisadoes commented on .github/workflows/pull-request.yml:
```diff
@@ -272,35 +272,78 @@ jobs:
+            exit 1
+          fi
+          npx graphql-inspector validate schema.graphql "src/**/*.graphql"
+  Graphql-Inspector:
```
Add a space between this and the previous job for readability

palisadoes reviewed (changes requested):
See comments

Priyanshuthapliyal2005 commented:
@coderabbitai full review

palisadoes commented:
@Priyanshuthapliyal2005 

1. The test may fail because of the many schema changes in the `develop-postgres` branch of the API and Admin repos.
2. We may have to create a schema.graphql in this repo that matches the one in the API.
3. This may assist in the migration of this repo to compatibility with the new API

I have some questions.

1. Do you think this is advisable?
2. What screens in this repo "would" and "would not" work generally speaking?

Priyanshuthapliyal2005 commented:
> @Priyanshuthapliyal2005 
> 
> 1. The test may fail because of the many schema changes in the `develop-postgres` branch of the API and Admin repos.
> 2. We may have to create a schema.graphql in this repo that matches the one in the API.
> 3. This may assist in the migration of this repo to compatibility with the new API
> 
> I have some questions.
> 
> 1. Do you think this is advisable?
> 2. What screens in this repo "would" and "would not" work generally speaking?

Yeah that’s a great idea! We can use a similar schema.graphql here. Due to the changes in the develop-postgres branch, the repo's coverage on Codecov is currently low. Mainly, the screens that rely heavily on MongoDB directly will not work, while others should function as expected.

palisadoes commented:
1. Please do that.
2. In the long run, how can we ensure that the schemas are always in sync?

Priyanshuthapliyal2005 commented:
> 1. Please do that.
> 2. In the long run, how can we ensure that the schemas are always in sync?

Should I create a new issue and PR for this, or continue making the changes here?

palisadoes commented:
This one

Priyanshuthapliyal2005 commented:
> This one

I have replaced the current schema.graphql with the one from the develop-postgres branch of talawa-api, commented out the clone step for talawa-api since it's no longer needed when using the schema directly, and updated the schema path accordingly.

Priyanshuthapliyal2005 commented:
https://github.com/coderabbitai full review

palisadoes commented:
Closing. Inactivity
